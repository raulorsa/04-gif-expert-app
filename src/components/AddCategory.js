import React, {useState} from 'react'

import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setinputValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length>2){

            //setCategorias( cats => [...cats, inputValue ] );
            setCategorias( cats => [inputValue,...cats ] );

            setinputValue('');
        }
        //console.log('Submit');

        
    }

    return (
        
        <form onSubmit={handleSubmit}>
            {/* <p> {inputValue}</p> */}

            <input 
            type="text"
            value= {inputValue}
            onChange= {handleInputChange}
            
            />
            
        </form>

    )
}

AddCategory.propTypes ={
    setCategorias: PropTypes.func.isRequired
}
