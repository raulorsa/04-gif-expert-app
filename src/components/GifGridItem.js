import React from 'react'

import PropTypes from 'prop-types';


export const GifGridItem = ( {id, title, url} ) => {
    console.log(title);

    return (
        <div className="card">
            <img src={ url} alt={title}/>
            <p> {title}</p>
        </div>
    )
}

GifGridItem.propTypes ={ 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,

}


/*
    1.Enzime
    2.Enzime to Json
    mostrar componentes
    *shallow
    *wrapper
         */
