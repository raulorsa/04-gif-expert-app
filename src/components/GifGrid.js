import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
//import React, {useState, useEffect}  from 'react'
import { GifGridItem } from './GifGridItem';
//import { getGift } from '../helpers/getGifs';

export const GifGrid = ({ category}) => {

    // const [images, setImages] = useState( [])

    const {data, loading} = useFetchGifs( category);

    // useEffect(() =>{
    //     getGift( category).then( imgs => setImages(imgs));
    // },[ category])




    return (
        <>
            <h3  > {category} </h3>

            {loading && <p>Loading</p> }

            <div className="card-grid animate__animated animate__fadeInDown">

                {
                   //console.log(data)
                   // <li key={img.id}> {img.title}</li>
                   
                       data.map( imgs => (

                        <GifGridItem 
                        key={imgs.id}
                        {...imgs}
                        />
                    ))
                }

            </div> 
        </>
    )
}
