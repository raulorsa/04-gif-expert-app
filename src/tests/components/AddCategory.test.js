import PropTypes from 'prop-types';
import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Pruebas en <GifGridItem />',() => {

    const setCategorias = jest.fn();
    let wrapper = shallow( <AddCategory setCategorias={ setCategorias} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategorias={ setCategorias} />);

    })

    test('debe mostrarse correctamente', () => {
        expect( wrapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo'

        input.simulate('change', {target:{
            value: value
        }});

        //expect(wrapper.find('p').text().trim()).toBe(value);

    })


    test('No Debe postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault (){} });

        expect( setCategorias ).not.toHaveBeenCalled();
    })


    test('debe de llamar el setCategorias y limpiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        const vacio = '';
        
        //Simular el inputChange
        input.simulate('change', {target:{ value: value }});


        //simular el submit
        wrapper.find('form').simulate('submit', {preventDefault (){} });

        //setCategorias de debe haber llamado
        expect( setCategorias ).toHaveBeenCalled();
        //Se ha llamado una sola vez
        expect( setCategorias ).toHaveBeenCalledTimes(1);

        //el valor del input debe vaciarse
        expect(wrapper.find('input').prop('value')).toBe('');


    })
    

})