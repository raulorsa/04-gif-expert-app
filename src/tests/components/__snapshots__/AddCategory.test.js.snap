// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
